### ISSUE TITLE
TrustEveryoneFactory in chat?

### ISSUE THREAD

**Issue by micheljung** [FAForever/downlords-faf-client#2161]
Just hopping in every now and then to see what you guys are doing :-)

For instance, why would we disable SSL verification and not even comment it in the code?

https://github.com/FAForever/downlords-faf-client/blob/6d1277017e3ed48e559f4d993a316ce4765e6999/src/main/java/com/faforever/client/chat/TrustEveryoneFactory.java#L11

I'm pretty sure this isn't a smart move ;)

**Comment by Sheikah45** [comment]
This was introduced in the change from PircBotX to KittehIRC and was used to replicate the configuration int PircBotx as seen here https://github.com/FAForever/downlords-faf-client/blob/7262cedf49eaa15fdb65186350641dfb8c55c789/src/main/java/com/faforever/client/chat/PircBotXChatService.java#L393

The purpose being to maintain the current functionality in the change over as kitteh connects over SSL by default.

**Comment by Sheikah45** [comment]
I believe the trust all certificates was originally used due to certificate issues with the irc server but am not certain and this change mainly was focused on converting the service to similar functionality. However the service can be changed but would require input from Brutus or if you know as from my investigation the SSL was disabled since PircBotX was first implemented.

**Comment by Brutus5000** [comment]
Since our reverse proxy supports raw TCP streams we could use TLS if we want to

**Comment by micheljung** [comment]
With the chat sending passwords over the wire, (secure) TLS would surely be preferred. Oh, wait, we don't need to because we `sha256(md5(md5(password)))` right? ;D

**Comment by Sheikah45** [comment]
yeah we currently still do perform that hashing acrobatics which would be nice to simplify

**Comment by Sheikah45** [comment]
i just need to know from Brutus how the irc has its TLS configured

**Comment by Sheikah45** [comment]
No longer needed now that chat is websockets
